src/bytes_mut.rs

```diff
@@ -819,6 +819,79 @@ impl BytesMut {
         }
     }
 
+    /// Attempts to reclaim the whole allocation of the `BytesMut`.
+    ///
+    /// If the `BytesMut` is empty but the underlying storage has been used before,
+    /// it might be possible to cheaply reclaim capacity as long as there are no
+    /// other `Bytes` or `BytesMut` handles alive.
+    ///
+    /// Returns `true` if the allocation was reclaimed (or reclaiming was not
+    /// necessary, because the underlying storage was empty); returns `false` if the
+    /// `BytesMut` is not empty or there are other outstanding references to the
+    /// underlying storage.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use bytes::BytesMut;
+    ///
+    /// let mut buf = BytesMut::with_capacity(64);
+    /// assert_eq!(true, buf.try_reclaim());
+    ///
+    /// buf.extend_from_slice(b"abcd");
+    /// let mut split = buf.split();
+    /// assert_eq!(false, split.try_reclaim());
+    /// assert_eq!(false, buf.try_reclaim());
+    /// drop(buf);
+    /// assert_eq!(false, split.try_reclaim());
+    /// split.clear();
+    /// assert_eq!(true, split.try_reclaim());
```

Comment:
I would probably include some asserts on what the capacity is here. And add some empty lines as well.

Reply:
Updated